<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [1]:
# Membuat dataframes dari dictionary
import pandas as pd

df = pd.DataFrame({'negara': ['spanyol', 'perancis', 'jerman'], 'ibu kota': ['madrid', 'paris', 'berlin']})
print(df)


     negara ibu kota
0   spanyol   madrid
1  perancis    paris
2    jerman   berlin


<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [42]:
# Membaca data dan memasukannya kedalam suatu variabel
import numpy as np
import csv

with open('diamonds.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    for row in csv_reader:
        print(row)


['', 'carat', 'cut', 'color', 'clarity', 'depth', 'table', 'price', 'x', 'y', 'z']
['1', '0.23', 'Ideal', 'E', 'SI2', '61.5', '55', '326', '3.95', '3.98', '2.43']
['2', '0.21', 'Premium', 'E', 'SI1', '59.8', '61', '326', '3.89', '3.84', '2.31']
['3', '0.23', 'Good', 'E', 'VS1', '56.9', '65', '327', '4.05', '4.07', '2.31']
['4', '0.29', 'Premium', 'I', 'VS2', '62.4', '58', '334', '4.2', '4.23', '2.63']
['5', '0.31', 'Good', 'J', 'SI2', '63.3', '58', '335', '4.34', '4.35', '2.75']
['6', '0.24', 'Very Good', 'J', 'VVS2', '62.8', '57', '336', '3.94', '3.96', '2.48']
['7', '0.24', 'Very Good', 'I', 'VVS1', '62.3', '57', '336', '3.95', '3.98', '2.47']
['8', '0.26', 'Very Good', 'H', 'SI1', '61.9', '55', '337', '4.07', '4.11', '2.53']
['9', '0.22', 'Fair', 'E', 'VS2', '65.1', '61', '337', '3.87', '3.78', '2.49']
['10', '0.23', 'Very Good', 'H', 'VS1', '59.4', '61', '338', '4', '4.05', '2.39']
['11', '0.3', 'Good', 'J', 'SI1', '64', '55', '339', '4.25', '4.28', '2.73']
['12', '0.23', 'Ideal', 

IOPub data rate exceeded.
The Jupyter server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--ServerApp.iopub_data_rate_limit`.

Current values:
ServerApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
ServerApp.rate_limit_window=3.0 (secs)



Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 
- Series memiliki persamaan dengan list yaitu dapat dilakukan metode slicing, yaitu mengakses/indexing pada object series tersebut. Juga series memiliki persamaan dengan dictionary yaitu index yang dimiliki series berguna seperti keys pada dictionary. Sesuai perbedaan data dari list dan dictionary, series dengan list data memiliki urutan dan bisa dirubah sedangkan series dengan dictionary koleksi pasangan key yang digunakan untuk mengakses data.
  
- Perbedaan dari Series dan DataFrame ini terletak pada strukturnya dan juga attribute dan method-method yang mereka miliki, untuk perbedaan strukturnya adalah sebagai berikut :

Series adalah objek 1 dimensi
DataFrame adalah suatu objek 2 dimensi, mirip seperti tabel

In [2]:
# code here
x = pd.Series([80,70,84], index=['ana', 'ali', 'cici'])
print(x)

ana     80
ali     70
cici    84
dtype: int64


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [36]:
# code here
import pandas as pd

df = pd.read_csv('diamonds.csv')
#createDataFrame
type(df['price'].iloc[0])
baris_pertama_price = df['price'].iloc[0]
print(baris_pertama_price)
# hasil 326

baris_terakhir_color = df['color'].iloc[-1]
print(baris_terakhir_color)
# hasil D

326
D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!


Statistik deskriptif mencakup statistik yang merangkum tendensi sentral, penyebaran, dan bentuk distribusi kumpulan data.

Statistik deskriptif merupakan proses analisis statistik yang fokus kepada manejemen, penyajian, dan klasifikasi data. Dengan proses ini, data yang disajikan akan menjadi lebih menarik lebih mudah dipahami, dan mampu memberikan makna lebih bagi pengguna data.

Statistik deskriptif memberikan deskripsi data secara ringkas dari bagaimana bentuk distribusi datanya, sampai ukuran penyebaran dari data tersebut

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [40]:
# code here
df = pd.read_csv('diamonds.csv')
df.describe()

Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000      4.720000      2.910000  
50%        5.700000      5.710000      3.530000  
75%        6.540000      6.540000      4.040000  
max       10.740000     58.900000     31.800000

In [44]:
rerata = sum(df['price']) / len(df['price'])
print(rerata)

print(np.std(df['depth'], ddof=1))

print(max(df['carat']))

3932.799721913237
1.432621318833661
5.01


Jawab:
- Metode yang di gunakan adalah describe() hasil tampilan seperti diatas
- Nilai rata-rata dari column price  :  3932.799721913237
- Nilai standar deviasi dari column depth : 1.432621318833661
- Nilai maximum dari column carat : 5.01